code/lib/preview-api/src/modules/store/csf/portable-stories.ts

```diff
@@ -105,68 +109,61 @@ export function composeStory<TRenderer extends Renderer = Renderer, TArgs extend
     abortSignal: new AbortController().signal,
     step: (label, play) => story.runStep(label, play, context),
     canvasElement: globalThis?.document?.body,
-    context: null!,
     canvas: {},
     ...story,
+    context: null!,
+    mount: null!,
   };
 
   context.context = context;
+  context.mount = story.mount(context);
+  context.renderToCanvas = async () => {
+    // Consolidate this renderContext with Context in SB 9.0
+    const unmount = await story.renderToCanvas?.(
+      {
+        componentId: story.componentId,
+        title: story.title,
+        id: story.id,
+        name: story.name,
+        tags: story.tags,
+        showError: (error) => {},
+        showException: (error) => {},
+        forceRemount: true,
+        storyContext: context,
+        storyFn: () => story.unboundStoryFn(context),
+        unboundStoryFn: story.unboundStoryFn,
+      } as RenderContext<TRenderer>,
+      context.canvasElement
+    );
+    if (unmount) {
+      cleanups.push(unmount);
+    }
+  };
 
-  const playFunction = story.playFunction
-    ? async (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
-        Object.assign(context, extraContext);
-        return story.playFunction!(context);
-      }
-    : undefined;
-
-  let previousCleanupsDone = false;
+  const playFunction = (extraContext?: Partial<StoryContext<TRenderer, Partial<TArgs>>>) => {
```

Comment:
This is more of an FYI, not sure what to do about this:

I remember doing this wrapper function so play was always defined, and people complained because they had some sort of checks in their portable stories tests that did things differently whether the story had a play function or not, and they were upset because they couldn't do such checks anymore.

So we can either:
- Do nothing, and wait for people to complain again
- Add a `hasPlayFunction` or similar property, to allow them to still do their checks

Reply:
let's make this backwards compatible as well